source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java

```diff
@@ -114,8 +128,285 @@ static void setup() {
+        String[] testSoundPath = new String[]{"sounds/tiger-roar.wav"};
+        String[][] testHintText = new String[][]{
+                {"Welcome to Animal Kingdom!", "I am Lenny the Lion.", "/cWhich tip do you wanna hear about?/s01What do potions do???/s02How to beat the final boss/s03Nothing. Bye"},
+                {"Potions heals you by (n) HP!", "I hope this helped."},
+                {"Final boss?? That Kangaroo??", "idk"},
+                {"Good luck!"}
+        };
+        NPCFactory.initiateDialogue(testSoundPath, testHintText);
+        verify(dialogueBoxService).updateText(testHintText, -2);
+
+//        for (String soundPath : testSoundPath) {
+//            AudioManager.playSound(soundPath);
+//            verify(resourceService).getAsset(soundPath, Sound.class);
+//        }
+    }
+
+    /**
+     * Tests the initialization of a friendly chicken by checking its creation, name, type,
+     * and the presence of necessary components.
+     */
+    @Test
+    void testChickenInitialisation() {
+        // Test creation of the friendlyChicken
+        Assertions.assertNotNull(friendlyChicken, "Chicken should not be null.");
+
+        // Test that the friendlyChicken has the correct name
+        String name = configs.friendlyChicken.getAnimalName();
+        Assertions.assertEquals("Chicken", name);
+
+        // Test that the friendlyChicken is an Entity
+        Assertions.assertEquals(Entity.class, friendlyChicken.getClass());
+
+        // Test that the friendlyChicken has a PhysicsComponent
+        Assertions.assertNotNull(friendlyChicken.getComponent(PhysicsComponent.class));
+
+        // Test that the friendlyChicken has a PhysicsMovementComponent
+        Assertions.assertNotNull(friendlyChicken.getComponent(PhysicsMovementComponent.class));
+
+        // Test that the friendlyChicken has a ColliderComponent
+        Assertions.assertNotNull(friendlyChicken.getComponent(ColliderComponent.class));
+
+        // Test that the friendlyChicken has a ConfigComponent
+        Assertions.assertNotNull(friendlyChicken.getComponent(ConfigComponent.class));
+    }
+
+    /**
+     * Tests that the friendly chicken has the correct sound path.
+     * //TODO:Change when sound path is updated
+     */
+    @Test
+    void TestChickenHasCorrectSoundPath() {
+        String[] sound = configs.friendlyChicken.getSoundPath();
+        Assertions.assertNotNull(sound);
+        Assertions.assertArrayEquals(new String[]{"sounds/chicken.wav"}, sound);
+    }
+
+    /**
+     * Tests that the friendly chicken has the correct base hint.
+     */
+    @Test
+    void TestChickenHasCorrectBaseHint() {
+        String[][] baseHint = configs.friendlyChicken.getBaseHint();
+        assertNotNull(baseHint);
+        Assertions.assertArrayEquals(new String[][]{{"Welcome to Animal Kingdom!","I am Charlie the Chicken."}},
+                baseHint);
+    }
+
+    /**
+     * Tests that the friendly chicken has a wait animation.
+     */
+//    @Test
```

Comment:
Why is this test commented out? Please either make it work with the current game or remove it.